flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -547,7 +552,7 @@ class TaskManager(
 
         // successful registration. associate with the JobManager
         // we disambiguate duplicate or erroneous messages, to simplify debugging
-        case AcknowledgeRegistration(_, leaderSessionID, jobManager, id, blobPort) =>
+        case AcknowledgeRegistration(_, leaderSessionID, jobManager, id, blobPort, jobManagerID) =>
```

Comment:
This does not work because a `TaskManager's` registration can also be acknowledged by an `AlreadyRegistered` message.